lib/assets/javascripts/turbograft/turbohead.coffee

```diff
@@ -4,39 +4,56 @@ class window.TurboHead
   update: (successCallback, failureCallback) ->
     activeAssets = extractTrackedAssets(@activeDocument)
     upstreamAssets = extractTrackedAssets(@upstreamDocument)
-    {activeScripts, newScripts} = processScripts(activeAssets, upstreamAssets)
 
-    if hasScriptConflict(activeScripts, newScripts)
-      return failureCallback()
+    newScripts = upstreamAssets
+      .filter(filterForNodeType('SCRIPT'))
+      .filter(noMatchFor({attribute: 'src', inCollection: activeAssets}))
 
-    updateLinkTags(activeAssets, upstreamAssets)
-    updateScriptTags(@activeDocument, newScripts, successCallback)
+    newLinks = upstreamAssets
+      .filter(filterForNodeType('LINK'))
+      .filter(noMatchFor({attribute: 'href', inCollection: activeAssets}))
 
-updateLinkTags = (activeAssets, upstreamAssets) ->
-  activeLinks = activeAssets.filter(filterForNodeType('LINK'))
-  upstreamLinks = upstreamAssets.filter(filterForNodeType('LINK'))
-  remainingActiveLinks = removeStaleLinks(activeLinks, upstreamLinks)
-  reorderedActiveLinks = reorderActiveLinks(remainingActiveLinks, upstreamLinks)
-  insertNewLinks(reorderedActiveLinks, upstreamLinks)
+    if newScripts.concat(newLinks).some(hasAssetConflicts(activeAssets))
+      return failureCallback()
```

Comment:
More to simulate passing an inlined named function (coffeescript doesn't let you), but ok.